src/test_workflow/benchmark_test/benchmark_test_cluster.py

```diff
@@ -0,0 +1,158 @@
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+        with open(self.output_file, "r") as read_file:
+            load_output = json.load(read_file)
+            print(load_output[self.stack_name])
+            self.create_endpoint(load_output)
+        self.wait_for_processing()
+
+    def create_endpoint(self, cdk_output: dict) -> None:
+        loadbalancer_url = cdk_output[self.stack_name].get('loadbalancerurl', None)
+        if loadbalancer_url is None:
+            raise RuntimeError("Unable to fetch the cluster endpoint from cdk output")
+        self.cluster_endpoint = loadbalancer_url
+        self.cluster_endpoint_with_port = "".join([loadbalancer_url, ":", str(self.port)])
+
+    @property
+    def endpoint(self) -> str:
+        return self.cluster_endpoint
+
+    @property
+    def endpoint_with_port(self) -> str:
+        return self.cluster_endpoint_with_port
+
+    @property
+    def port(self) -> int:
+        return 80 if self.args.insecure else 443
+
+    def terminate(self) -> None:
+        command = f"cdk destroy {self.stack_name} {self.params} --force"
+        logging.info(f'Executing "{command}" in {os.getcwd()}')
+
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+
+    def wait_for_processing(self, tries: int = 3, delay: int = 15, backoff: int = 2) -> None:
+        logging.info(f"Waiting for domain at {self.endpoint} to be up")
+        protocol = "http://" if self.args.insecure else "https://"
+        url = "".join([protocol, self.endpoint, "/_cluster/health"])
+        request_args = {"url": url} if self.args.insecure else {"url": url, "auth": HTTPBasicAuth("admin", "admin"), "verify": False}  # type: ignore
+        retry_call(requests.get, fkwargs=request_args,
+                   tries=tries, delay=delay, backoff=backoff)
+
+    def setup_cdk_params(self, config: dict) -> dict:
+        if self.args.stack_suffix:
+            suffix = self.args.stack_suffix + '-' + self.manifest.build.id + '-' + self.manifest.build.architecture
+        else:
+            suffix = self.manifest.build.id + '-' + self.manifest.build.architecture
+        return {
+            "distributionUrl": self.manifest.build.location,
+            "vpcId": config["Constants"]["VpcId"],
+            "account": config["Constants"]["AccountId"],
+            "region": config["Constants"]["Region"],
+            "suffix": suffix,
+            "securityDisabled": "true" if self.args.insecure else "false",
+            "cpuArch": self.manifest.build.architecture,
+            "singleNodeCluster": "true" if self.args.singleNode else "false",
+            "distVersion": self.manifest.build.version,
+            "minDistribution": "true" if self.args.minDistribution else "false",
+            "serverAccessType": config["Constants"]["serverAccessType"],
+            "restrictServerAccessTo": config["Constants"]["restrictServerAccessTo"],
+            "additionalConfig": self.args.additionalConfig,
```

Comment:
There's a lot of pass through params. I wonder whether we're better off not exposing all these command line options and only accepting a config file and passing it through here?

Reply:
Could you please give an example here, not sure if I understood the comment properly. 
@dblock

Reply:
I think I'm saying that almost all `Count` arguments come from args. I don't know whether that's the right answer, but a config file that contained all these options could be passed as 1 single argument `--config foobar.yml`. Then this code doesn't need to know about `dataNodeCount` for example, that would be contained inside `foobar.yml`.

Reply:
The idea to keep them as args is that there may be cases where a user wants to try different cluster configurations for running benchmark in parallel with different node counts for each cluster. Another is the ML use case where a user wants to have ML node enabled. 
@dblock

Reply:
That's fine. I think we're just arguing command line params vs. a config file.